core/pluginconfigreaderwitherror.go

```diff
@@ -236,29 +236,82 @@ func (reader PluginConfigReaderWithError) GetPluginArray(key string, defaultValu
 
 // GetModulatorArray returns an array of modulator plugins.
 func (reader PluginConfigReaderWithError) GetModulatorArray(key string, logScope tlog.LogScope, defaultValue ModulatorArray) (ModulatorArray, error) {
-	modPlugins, err := reader.GetPluginArray(key, []Plugin{})
+	modulators, _, _, err := reader.getModulatorArrays(key, logScope)
 	if err != nil {
 		return defaultValue, err
 	}
-	if len(modPlugins) == 0 {
+	if modulators == nil {
+		return defaultValue, nil
+	}
+
+	return modulators, nil
+}
+
+// GetFilterArray returns an array of filter plugins.
+func (reader PluginConfigReaderWithError) GetFilterArray(key string, logScope tlog.LogScope, defaultValue FilterArray) (FilterArray, error) {
+	_, filters, _, err := reader.getModulatorArrays(key, logScope)
+	if err != nil {
+		return defaultValue, err
+	}
+	if filters == nil {
 		return defaultValue, nil
 	}
 
+	return filters, nil
+}
+
+// GetFormatterArray returns an array of formatter plugins.
+func (reader PluginConfigReaderWithError) GetFormatterArray(key string, logScope tlog.LogScope, defaultValue FormatterArray) (FormatterArray, error) {
+	_, _, formatters, err := reader.getModulatorArrays(key, logScope)
+	if err != nil {
+		return defaultValue, err
+	}
+	if formatters == nil {
+		return defaultValue, nil
+	}
+
+	return formatters, nil
+}
+
+// getModulatorArrays returns ModulatorArray, FilterArray and FormatterArray arrays
+func (reader PluginConfigReaderWithError) getModulatorArrays(key string, logScope tlog.LogScope) (ModulatorArray, FilterArray, FormatterArray, error) {
```

Comment:
I would rather split this in 3 separate functions.
1. Clearer, shorter code + less return variables
2. Stricter typing / error handling (filter arrays must not contain formatters)
3. Code duplication is so minor it can be neglected